ckan/lib/munge.py

```diff
@@ -119,6 +127,34 @@ def munge_filename(filename):
     return filename
 
 
+def munge_filename(filename):
+    ''' Tidies a filename
+
+    Keeps the filename extension.
+    Strips off any path on the front.
+    It can be useful to keep the extension (e.g. .csv) in a resource when the
+    format field set.
```

Comment:
The reason we need the extension is for the datapusher to function etc.

It is needed when the format is __not__ set